onnxruntime/core/session/environment.cc

```diff
@@ -329,4 +335,26 @@ Internal copy node
   return status;
 }
 
+Status Environment::CreateAndRegisterExecutionProvider(bool use_arena, std::string provider_type, int* provider_global_index) {
+  *provider_global_index = -1;
+  std::unique_ptr<IExecutionProvider> ep;
+  if (provider_type == kCpuExecutionProvider) {
+    ep = onnxruntime::CPUProviderFactoryCreator::Create(use_arena)->CreateProvider();
+  } else if (provider_type == kXnnpackExecutionProvider) {
```

Comment:
>else if (provider_type == kXnnpackExecutionProvi [](http://example.com/codeflow?start=4&length=48)

this probably not scale to all the EPs, could we leverage the code how we create EPs in InferenceSession?